internal/pkg/term/selector/selector.go

```diff
@@ -251,7 +251,9 @@ func (s *DeploySelect) DeployedService(prompt, help string, app string, opts ...
 	var deployedSvc DeployedService
 	if len(svcEnvNames) == 1 {
 		deployedSvc = svcEnvs[svcEnvNames[0]]
-		log.Infof("Only found one deployed service %s in environment %s\n", color.HighlightUserInput(deployedSvc.Svc), color.HighlightUserInput(deployedSvc.Env))
+		if s.svc == "" {
+			log.Infof("Only found one deployed service %s in environment %s\n", color.HighlightUserInput(deployedSvc.Svc), color.HighlightUserInput(deployedSvc.Env))
+		}
 		return &deployedSvc, nil
```

Comment:
```suggestion
              if s.svc != "" && s.env != "" {
                      return &deployedSvc, nil
              }
		log.Infof("Only found one deployed service %s in environment %s\n", color.HighlightUserInput(deployedSvc.Svc), color.HighlightUserInput(deployedSvc.Env))
		return &deployedSvc, nil
```
I feel this makes more sense to me, since the combination is only unique when both `--name` and `--env` are specified. Otherwise, users won't be surprised to see this message since the combination is not unique and they'd like to know what values for `env` and `svc` we set by default.

Reply:
What do you think about this? 
```
	if len(svcEnvNames) == 1 {
		deployedSvc = svcEnvs[svcEnvNames[0]]
		// if either --name or --env flag set
		if (s.svc == "" && s.env != "") || (s.svc != "" && s.env == "") {
			log.Infof("Service %s found in environment %s\n", color.HighlightUserInput(deployedSvc.Svc), color.HighlightUserInput(deployedSvc.Env))
		} else if s.svc == "" {
			log.Infof("Only found one deployed service %s in environment %s\n", color.HighlightUserInput(deployedSvc.Svc), color.HighlightUserInput(deployedSvc.Env))
		}
		return &deployedSvc, nil
	}
```

Reply:
Yeah I like it. Just to simply the logic:
``` go
	if len(svcEnvNames) == 1 {
		deployedSvc = svcEnvs[svcEnvNames[0]]
                if s.svc == "" && s.env == "" {
			log.Infof("Only found one deployed service %s in environment %s\n", color.HighlightUserInput(deployedSvc.Svc), color.HighlightUserInput(deployedSvc.Env))
                }
                if (s.svc != "") != (s.env != "") {
			log.Infof("Service %s found in environment %s\n", color.HighlightUserInput(deployedSvc.Svc), color.HighlightUserInput(deployedSvc.Env))
                }
		return &deployedSvc, nil
	}
```